hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/BucketEndpoint.java

```diff
@@ -709,7 +710,10 @@ private void addKey(ListObjectResponse response, OzoneKey next) {
     keyMetadata.setKey(EncodingTypeObject.createNullable(next.getName(),
         response.getEncodingType()));
     keyMetadata.setSize(next.getDataSize());
-    keyMetadata.setETag("" + next.getModificationTime());
+    String eTag = next.getMetadata().get(ETAG);
+    if (eTag != null) {
+      keyMetadata.setETag(eTag);
```

Comment:
Should we standardize so that the ETag returned needs to be surrounded by wrap in quotes? Similar to `CopyObjectResponse`? 

From https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html

However, it seems that the S3A pass even without the quotes. From what I briefly saw from `AbstractContractEtagTest`, the test does not assert the quotations of the ETag.

Reply:
In the AWS page it is interesting that both `Key` and `ETag` are defined as `string`, but only that latter has quotes as part of the value.

Reply:
From what I understand the ETag is specified by RFC to always be surrounded by double quotes.

Ref: https://github.com/aws/aws-sdk-go-v2/issues/1696

@vtutrinov Could you help confirm this?

Reply:
Thanks for the pointers.  Updated to wrap in quotes.